### PR TITLE
feat: Batch update

### PR DESCRIPTION
**Pull request by kakoc**
batch-loading
saving-succ
saving-err

Used stubs for Tasks and Assignments parts(they are not implemented; they will be implemented in other tasks, how i understood). So, when they will be implemented need make integration and write tests

### CODE REVIEW

File: src/app/containers/BatchUpdate/index.tsx [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+                },
+                { Header: 'Task columns', accessor: 'header', width: 400, style: { alignSelf: cellStyle.alignSelf } },
+                {
+                    Header: 'No import columns',
+                    accessor: 'checkbox',
+                    minWidth: 200,
+                    style: cellStyle,
+                },
+                { Header: 'Select field', accessor: 'dropdown', style: cellStyle, minWidth: 300 },
+            ]}
```
Comment (SIARHEI-SMANTSAR) [inline src/app/containers/BatchUpdate/index.tsx]: Create constant for table config

File: src/app/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import { ButtonDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
+
+import { classNames } from 'core/styles';
+
+const cn = classNames(require('./index.scss'));
+
+class Dropdown extends React.Component<any, any> {
```
Comment (SIARHEI-SMANTSAR) [inline src/app/components/Dropdown/Dropdown.tsx]: Please add types

File: src/app/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    select = (item: any) => {
+        this.setState({ currentValue: item.value });
+        this.props.onSelect(item);
+    };
+
+    render() {
+        return (
+            <ButtonDropdown
+                isOpen={this.state.isOpen}
+                toggle={() => this.setState((prevState: any) => ({ isOpen: !prevState.isOpen }))}
```
Comment (SIARHEI-SMANTSAR) [inline src/app/components/Dropdown/Dropdown.tsx]: Don't use arrow functions into render method

File: src/app/containers/BatchUpdate/index.tsx [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+
+const mapStateToProps = (state: RootState, props: any): any => {
+    return {
+        ...props,
+        courses: coursesSelectors.getCoursesNames(state.courses.data),
+        tasks: state.tasks.courseRelated,
+    };
+};
+
+const mapDispatchToProps = (dispatch: any, props: any): any => {
```
Comment (Shastel) [inline src/app/containers/BatchUpdate/index.tsx]:
```
[mapDispatchToProps(dispatch, [ownProps]): dispatchProps] (Object or Function): If an object is passed, each function inside it is assumed to be a Redux action creator. An object with the same function names, but with every action creator wrapped into a dispatch call so they may be invoked directly, will be merged into the component’s props.
```

File: src/app/core/validation/batchUpdate/index.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { Roles } from 'core/models/user';
+import { AssignmentsType } from 'core/models';
+import { checkers, prepareForChecking, isAllNeedData, checkTable, toVerifiableData, checkFor } from './dataChecking';
+import { matchUsers } from 'core/api';
+import { makeErrorsList, messageMaker } from '../utils';
+
+const dataRetriever = (...what: any[]) => (data: any) =>
+    what.reduce((needRetrievedData: any[], needValue: any) => needRetrievedData.concat(data[needValue]), []);
```
Comment (Shastel) [inline src/app/core/validation/batchUpdate/index.ts]: If i got the point you can do same thing easily with `map`

File: src/app/containers/BatchUpdate/index.tsx (hunk 4 above)
Comment (kakoc) [reply src/app/containers/BatchUpdate/index.tsx]: done.

File: src/app/core/validation/batchUpdate/index.ts (hunk 5 above)
Comment (kakoc) [reply src/app/core/validation/batchUpdate/index.ts]: done)

**Pull request closed without merging**